src/inmanta/protocol/methods_v2.py

```diff
@@ -1357,3 +1357,73 @@ def set_password(username: str, password: str) -> None:
     :raises NotFound: Raised when the user does not exist
     :raises BadRequest: Raised when server authentication is not enabled
     """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    validate_sid=False,
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def unmanaged_resource_create(env: uuid.UUID, unmanaged_resource_id: str, values: Dict[str, str]) -> None:
+    """
+    create a discovered resource in the DB
+    :param env: The id of the environment this resource belongs to
+    :param unmanaged_resource_id: The id of the unmanaged resource
+    :param values: The values associated with the unmanaged_resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/",
+    validate_sid=False,
+    operation="POST",
+    agent_server=True,
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.agent],
+    api_version=2,
+)
+def unmanaged_resource_create_batch(env: uuid.UUID, unmanaged_resources: List[UnmanagedResource]) -> None:
+    """
+    create multiple discovered resource in the DB
+    :param env: The id of the environment this resource belongs to
+    :param unmanaged_resources: List of UnmanagedResource containing the unmanaged_resource_id and values for each resource
+    """
+
+
+@typedmethod(
+    path="/unmanaged/<unmanaged_resource_id>",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def unmanaged_resources_get(tid: uuid.UUID, unmanaged_resource_id: str) -> model.UnmanagedResource:
+    """
+    Get a single discovered resource
+    :param tid: the id of the environment in which to get the unmanaged resource.
```

Comment:
I seem to recall that the documentation generation tool requires an empty line between the description and parameters. I am not sure, but it doesn't hurt to add the empty line anyway. 
```suggestion
    Get a single discovered resource.

    :param tid: the id of the environment in which to get the unmanaged resource.
```